rfcs/20180907-contrib-sunset.md

```diff
@@ -0,0 +1,280 @@
+| periodic_resample  |                  | delete (no owner)                    |
+| pi_examples        |  petewarden      | delete (will need new examples)      |
+| predictor          | ispirmustafa karmel | delete (replaced by tfhub)        |
+| proto              | jsimsa ebrevdo   | move to core                         |
+| quantization       |  petewarden      | delete (absorbed into tflite)        |
+| quantize           | suharshs         |                                      |
+| rate               |  itsmeolivia     |                                      |
+| receptive_field    |                  |                                      |
+| recurrent          | drpngx zffchen78 | replaced with new RNN API?           |
+| reduce_slice_ops   |                  | delete (no owner)                    |
+| remote_fused_graph |  satok16         |                                      |
+| resampler          | fabioviola kosklain | move to tensorflow/scientific?    |
+| rnn            | ebrevdo (scottzhu)   | replace with new RNN API             |
+| rpc                | ebrevdo jsimsa   |                                      |
+| saved_model        | karmel           | move to core                         |
+| seq2seq        | ebrevdo (scottzhu)   | adapt, move to core                  |
+| session_bundle     |                  | delete (replaced by SavedModel)      |
+| signal             | rryan            | move to core (replace existing) or tensorflow/scientific? |
+| slim               |  sguada          | move to tensorflow/models?           |
+| solvers            |  rmlarsen        | move to tensorflow/scientific?       |
+| sparsemax          |                  | move to tensorflow/contrib?          |
+| specs              |                  | delete                               |
+| staging            |                  | delete (redundant)                   |
+| stat_summarizer    |                  | delete (no owner)                    |
+| stateless          | (*ebrevdo *girving) | replace our random ops with this? |
+| summary            |  nickfelt        | move to core, replacing tf.summary   |
+| tensor_forest      | nataliaponomareva yupbank | delete (moving to core)     |
+| tensorboard        |  nickfelt  jart  | move to tensorflow/tensorboard       |
+| tensorrt           | tobyboyd *samikama *aaroey *jjsjann123| move essential parts to core |
+| testing            | ispirmustafa     | move to core, make private           |
+| text               |                  | move to tensorflow/contrib?          |
+| tfprof             |                  | delete (replaced by tf.profiler)     |
+| timeseries         | gsundeep karmel  | move to tensorflow/estimator         |
+| tpu                | saeta            | move to core                         |
+| training           | ebrevdo sguada joel-shor |                              |
+| util               |                  | delete (no owner), or move to tools  |
```

Comment:
There's one function in this project that proved to be particularly useful for static optimizations on libraries built on top on TensorFlow, and as far as I know there is no alternative: tf.contrib.util.constant_value.

Considering the future of this project is undecided yet, would it be possible to ensure that at least this particular function or an alternative to it are available in TensorFlow 2.0?

Reply:
I am not sure how interesting this function still is with eager execution. @mrry FYI.

Reply:
Eager execution in 2.0 still allows you to have graphs (though we'd prefer you think of them as functions), and those can still be exported, executed remotely, etc. `constant_value` is not particularly useful with eager execution since you can simply look at any tensor (its value will be there), or run any function (which will produce a value). 

Or is the capability you need the implicit estimate of whether the value can be computed cheaply?

@asimshankar

Reply:
Glad to read about graphs in 2.0. If you happen to have any links where I can learn more about this new design, I'll be very happy to take a look so I can make my comments more useful and targeted in the future.

Answering your question in TF 1.x terms, we need the ability to quickly and statically access the value of constant tensors and statically known shapes without involving a session or a tensor evaluation. I'm not fully sure how this would translate to TF 2.0, but we need something faster than tf.Tensor.eval().

Reply:
I'm hoping to publish a proposal for eager and graphs in TF 2.0 soon (I'm working on the text, PR hopefully this week). See keep and eye out for that.

Reply:
@asimshankar 
Thanks, I'll definitely keep an eye.

@martinwicke 
I think we're on the same page here, but let me confirm.

"I believe for Tensors which you know are constants, the value information
is simply always available, same for shapes, so I do think this function is
redundant."

Is that the case too if I don't know if a tensor is constant? For example, if I'm writing some function and I get a tensor, can I simply try to "statically" get its value? (which I expect would work if it's constant or somehow statically evaluable, but fail otherwise)

From what you say it looks like I probably can, and if that's the case then this function would indeed become redundant, but just making sure.

Reply:
@martinwicke 
Just to not leave this pending, will it be possible to check if a given tensor is constant in TF 2.x? That should also help solving the problem. In 1.x you could check the type of the op the tensor belongs to, but I ignore if that will be possible in TF 2.x.

Reply:
(FYI, re https://github.com/tensorflow/community/pull/18#discussion_r218270083 see #20)